### ISSUE TITLE
Run applications in (Linux) console

### ISSUE THREAD

**Issue by excitoon** [joncampbell123/dosbox-x#3968]
### Is your feature request related to a problem? Please describe.

Hey. I'd like to have possibility to run console applications without any windows and having their output (properly) redirected to host machine.

### What you want

Hey. I'd like to have possibility to run console applications without any windows and having their output (properly) redirected to host machine.

### Have you checked that no similar feature request(s) exist?

- [X] I have searched and didn't find any similar feature request.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by Torinde** [comment]
Yes, that would be useful also for Windows and other host platforms. Related Console features:
- integrate in Windows Terminal microsoft/terminal/issues/3558 :
> If DOSBox can be run as a strictly text-mode application, it can be hosted in Terminal perfectly fine.
Can at least the DOS shell run in text mode? Will the video window be spawned outside Terminal?
- you can start DOS applications from yours favorite console file manager: [Far](http://www.farmanager.com/index.php?l=en), [DN2](http://www.dnosp.com/), cmd, or directly from [ConEmu](https://conemu.github.io/en/DosBox.html)
- DOSBox-X as wrapper around DOS executables, receiving input from and returning output to Win32 console https://github.com/joncampbell123/dosbox-x/issues/1184#issuecomment-513803516
- #1838 https://github.com/joncampbell123/dosbox-x/issues/1184#issuecomment-524695061
- suggestions as you type (for commands, parameters, files/folders, …) and other ideas from [Yori](http://www.malsmith.net/yori/)
- clipboard copy text from DOSbox -> paste in Windows together with the letters and background color (or even animated GIF for ANSI/4DOS blinking letters) - and vice versa (colored/blinking letters/background from Windows -> pasted in DOSbox)
- **Scroll (e.g. go back to see text that flew upwards too fast - instead of |more) - via the scroll bar that is host controllable**
- Find (both on-screen and in-history/back-scroll)
- settings equivalent to cmd.exe Properties\Options, Font, Layout, Colors, Terminal

**Comment by excitoon** [comment]
@Torinde Yeah, all of these are related and many can be solved simply by redirecting output. I guess I'll make this one as I need `DEBUG.COM` output so much.

**Comment by excitoon** [comment]
This is a stack of junk, however, look at `0x21/0x02` and `0x21/0x09` implementations:
```
        case 0x09:      /* Write string to STDOUT */
            {
                uint8_t c;uint16_t n=1;
                PhysPt buf=SegPhys(ds)+reg_dx;
                std::string str="";
                if (mem_readb(buf)=='T')
                    while ((c=mem_readb(buf++))!='$') {
                        str+=std::string(1, c);
                        if (str.length()>42) break;
                    }
                buf=SegPhys(ds)+reg_dx;
                reqwin=(str.length()==42&&!strncmp(str.c_str(),"This program cannot be run in DOS mode.",39))||(!strncmp(str.c_str(),"This program requires Microsoft Windows.",40))||(str.length()==38&&!strncmp(str.c_str(),"This program must be run under Win32",36));
                if (!winautorun||!reqwin||control->SecureMode()||!startquiet)
                    while ((c=mem_readb(buf++))!='$')
                        DOS_WriteFile(STDOUT,&c,&n);
                reg_al=c;
            }
```
And single characters skip all these checks:
```
        case 0x02:      /* Write character to STDOUT */
            {
                uint8_t c=reg_dl;uint16_t n=1;
                DOS_WriteFile(STDOUT,&c,&n);
                //Not in the official specs, but happens nonetheless. (last written character)
                reg_al=(c==9)?0x20:c; //strangely, tab conversion to spaces is reflected here
            }
            break;

```

**Comment by excitoon** [comment]
```
$ git add src/debug/debug_gui.cpp
The following paths are ignored by one of your .gitignore files:
src/debug
hint: Use -f if you really want to add them.
hint: Turn this message off by running
hint: "git config advice.addIgnoredFile false"
```

**Comment by excitoon** [comment]
- [ ] skip keyboard echo output
- [ ] remove blank lines at start
- [ ] remove "LOG: Logging output has been disabled."

**Comment by excitoon** [comment]
Feel free to make pull requests to my branch.
